framework/quota/src/main/java/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -558,6 +563,14 @@ protected BigDecimal getUsageValueAccordingToUsageUnitType(UsageVO usageRecord,
         }
     }
 
+    protected void setHoursInCurrentMonth() {
+        LocalDate currentDate = LocalDate.now();
```

Comment:
@stephankruggg is it possible that this calculation happens for just a minute ago/last month on the 1st of the next month. How will this be handled?

Reply:
@DaanHoogland, sorry for the delay. 
I have implemented some changes to address the issues you pointed out by getting the date directly from each record.